termux-exec.c

```diff
@@ -231,8 +236,33 @@ int execve(const char* filename, char* const* argv, char* const* envp)
 		}
 	}
 
-	int ret = real_execve(filename, argv, envp);
+	int ret = real_exec_fn(filename, argv, envp);
 	free(new_argv);
 	free(new_envp);
 	return ret;
+	
+}
+
+int _execvp(const char* filename, char* const* argv, char* const* envp)
+{
+        (void)envp;
+        int (*real_execvp) (const char*, char* const[]) = dlsym(RTLD_NEXT, "execvp");
```

Comment:
You should cache the `dlsym()` result. There's no point in calling `dlsym()` over and over again. It's too wasteful. Fix other similar places as well.

Reply:
The process (include .so) lifecycle only span one command run. So the cache will only get used once.